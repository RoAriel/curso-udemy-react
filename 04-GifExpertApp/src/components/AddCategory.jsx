import { useState } from 'react'

export const AddCategory = ({ onNewCategory }) => {

  const [inputValue, setInputValue] = useState('One Piece')

  const onInputChange = (event) => {

    setInputValue(event.target.value)
  }

  const onSubmit = (event) => {
    event.preventDefault()

    const newInputValue = inputValue.trim()

    if (newInputValue.length <= 1) return

    onNewCategory(newInputValue)
    setInputValue('')
 
  }

  return (
    <form onSubmit={ onSubmit }>
      <input type="text"
        placeholder="Buscar gif"
        value={ inputValue }
        onChange={ onInputChange } />
    </form>

  )
}