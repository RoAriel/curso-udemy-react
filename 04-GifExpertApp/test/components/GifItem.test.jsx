import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem"

describe('Pruebas en GiftItem', () => {

    const title = 'One Piece'
    const url = 'https://luffy-dono.com/'

    test('should match snapshot', () => {
    const {container} = render(<GifItem title={title} url={url}/>)

    expect(container).toMatchSnapshot()
  })

  test('should  mostar la img con el URL y el ALT indicado', () => {
    
    render(<GifItem title={title} url={url}/>)
    
    const {src, alt} = screen.getByRole('img')

    expect(src).toBe(url)
    expect(alt).toBe(title)

  })

  test('should mostrar el titulo', () => {
    
    render(<GifItem title={title} url={url}/>)

    expect(screen.getByText(title)).toBeTruthy();
    
  })
  
  
  
})
