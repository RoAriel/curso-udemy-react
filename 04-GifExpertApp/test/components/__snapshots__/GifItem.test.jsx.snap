// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GiftItem should match snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="One Piece"
      src="https://luffy-dono.com/"
    />
    <p>
      One Piece
    </p>
  </div>
</div>
`;
